medical/models/res_partner.py

```diff
@@ -44,6 +44,16 @@ class ResPartner(models.Model):
     )
 
     @api.multi
+    def _get_medical_entity(self):
+        self.ensure_one()
+        if not self.type:
```

Comment:
It's definitely short-circuiting here. Otherwise, the second part would throw an exception, which is what the original line was trying to prevent.